subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/matcher/TextMatchers.kt

```
@@ -0,0 +1,135 @@
+        if (stringMatcher.matches(text)) {
+            return true
+        }
+        mismatchDescription.appendText("view.getText() was ").appendValue(text.highlightNbsp())
+        if (textView.transformationMethod != null) {
+            val transformedText = textView.transformationMethod.getTransformation(text, textView)?.toString()
+            mismatchDescription.appendText(" transformed text was ")
+                .appendValue(transformedText?.highlightNbsp())
+            if (transformedText != null) {
+                return stringMatcher.matches(transformedText)
+            }
+        }
+        return false
+    }
+}
+
+internal class WithCharSequenceMatcher(
+    private val resourceId: Int,
+    private val method: TextViewMethod,
+) : BoundedDiagnosingMatcher<View, TextView>(TextView::class.java) {
+    private var resourceName: String? = null
+    private var expectedText: String? = null
+
+    internal enum class TextViewMethod {
+        GET_TEXT, GET_HINT
+    }
+
+    override fun describeMoreTo(description: Description) {
+        when (method) {
+            TextViewMethod.GET_TEXT -> description.appendText("view.getText()")
+            TextViewMethod.GET_HINT -> description.appendText("view.getHint()")
+        }
+        description.appendText(" equals string from resource id: ").appendValue(resourceId)
+        if (null != resourceName) {
+            description.appendText(" [").appendText(resourceName).appendText("]")
+        }
+        if (null != expectedText) {
+            description.appendText(" value: ").appendText(expectedText!!.highlightNbsp())
+        }
+    }
+
+    override fun matchesSafely(textView: TextView, mismatchDescription: Description): Boolean {
+        if (null == expectedText) {
+            try {
+                expectedText = textView.resources.getString(resourceId)
+            } catch (ignored: Resources.NotFoundException) {
+                /* view could be from a context unaware of the resource id. */
```

Maybe fail fast makes sense

---

I just copied it as is, so as to preserve original logic, which might be updated in Espresso lib later.
Should we leave it intact, shouldn't we?

---

Ok, I don't want to prolong your PR with this experiment